messaging/src/main/java/org/axonframework/messaging/eventhandling/processing/ProcessorEventHandlingComponents.java

```diff
@@ -145,4 +157,40 @@ public Set<Object> sequenceIdentifiersFor(@Nonnull EventMessage event, @Nonnull
                          .map(c -> c.sequenceIdentifierFor(event, context))
                          .collect(Collectors.toSet());
     }
+
+    /**
+     * Handles reset for all components that support it.
+     * <p>
+     * This method invokes the reset handler on each component that supports reset operations. All handlers must
+     * complete successfully for the operation to succeed.
+     *
+     * @param resetContext The reset context message.
+     * @param context      The processing context.
+     * @return A future that completes when all reset handlers have completed.
+     */
+    @Nonnull
+    public CompletableFuture<Void> handleReset(@Nonnull ResetContext resetContext,
+                                               @Nonnull ProcessingContext context) {
+        MessageStream<Message> result = MessageStream.empty();
+
+        for (var component : components) {
+            if (component.supportsReset()) {
+                result = result.concatWith(component.handle(resetContext, context).cast());
+            }
+        }
+
+        return result.ignoreEntries()
+                     .asCompletableFuture()
+                     .thenApply(v -> null);
```

Comment:
Super nit: Don't we have a `FutureUtils` method for this "return-void" behavior? If not, might be nice, for clarity.